### ISSUE TITLE
What's wrong in my Residual block implementation?

### ISSUE THREAD

**Issue by keunwoochoi** [fchollet/keras#1910]
I posted a residual block implementation in Keras here: https://github.com/keunwoochoi/residual_block_keras 
However it doesn't seem to work properly. 

I'm using Max-pooling as pooling size > kernel size sometimes, and current version also have dropouts layers. It didn't seem not working even without dropouts. The loss oscillates, huge gap between training loss and validation loss, etc. What would be the problem? It's quite difficult to spot one.

**Comment by cmishra** [comment]
Regarding what you posted on #1321:

- I think (1,1) convolution with stride > 1 works. It seems to be what the original authors did. Quote source is below, pulled from Section 3.3 the paragraph starting with 'Residual Network'. The emphasis is mine. 

> When the dimensions increase (dotted line shortcuts in Fig. 3), we consider two options: (A) The shortcut still performs identity mapping, with extra zero entries padded for increasing dimensions. This option introduces no extra parameter; (B) **The projection shortcut in Eqn.(2) is used to match dimensions (done by 1×1 convolutions). For both options, when the shortcuts go across feature maps of two sizes, they are performed with a stride of 2**. [source](http://arxiv.org/pdf/1512.03385v1.pdf)

- It should be easier to use, but I think we're thinking of different things. Let's discuss that after we have working minimal implementations.

Have you create any minimal examples that use your [residual_blocks.py](https://github.com/keunwoochoi/residual_block_keras/blob/master/residual_blocks.py) code without the complexity of [example.py](https://github.com/keunwoochoi/residual_block_keras/blob/master/example.py)?

**Comment by keunwoochoi** [comment]
Yes I'm going to update it today, probably with cifar or mnist.
It seems to be working well now, so it's more worth doing it. 
Thanks for letting me know (1x1), I yet don't understand how it works though. It's computationally quite demanding and don't have much time to run with different settings (average pooling vs mp, ..) but example code is coming soon!

**Comment by keunwoochoi** [comment]
I updated it with mnist example, you can just run `example.py`. But I didn't make any change on subsampling, there's just a minor change on the subsampling of shortcut - from `MaxPooling2D` to `AveragePooling2D`.

**Comment by sergeyf** [comment]
A more recent study: http://arxiv.org/pdf/1603.05027v1.pdf

**Comment by keunwoochoi** [comment]
@sergeyf I updated the code to follow up the new paper, FYI.
